zingolib/src/wallet/data.rs

```diff
@@ -343,8 +345,42 @@ impl PartialEq for OutgoingTxData {
     }
 }
 impl OutgoingTxData {
-    /// TODO: Add Doc Comment Here!
+    const VERSION: u64 = 0;
+    /// Before version 0, OutgoingTxData didn't have a version field
+    pub fn read_old<R: Read>(mut reader: R) -> io::Result<Self> {
+        let address_len = reader.read_u64::<LittleEndian>()?;
+        let mut address_bytes = vec![0; address_len as usize];
+        reader.read_exact(&mut address_bytes)?;
+        let address = String::from_utf8(address_bytes).unwrap();
+
+        let value = reader.read_u64::<LittleEndian>()?;
+
+        let mut memo_bytes = [0u8; 512];
+        reader.read_exact(&mut memo_bytes)?;
+        let memo = match MemoBytes::from_bytes(&memo_bytes) {
+            Ok(mb) => match Memo::try_from(mb.clone()) {
+                Ok(m) => Ok(m),
+                Err(_) => Ok(Memo::Future(mb)),
+            },
+            Err(e) => Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("Couldn't create memo: {}", e),
+            )),
+        }?;
+
+        Ok(OutgoingTxData {
+            recipient_address: address,
+            value,
+            memo,
+            recipient_ua: None,
+            output_index: None,
+        })
+    }
+
+    /// Read an OutgoingTxData from its serialized
+    /// representation
     pub fn read<R: Read>(mut reader: R) -> io::Result<Self> {
+        let _external_version = CompactSize::read(&mut reader);
```

Comment:
this variable. shouldnt be _, right? if the version doesnt exist, its not going to load correctly.

the fact that the owls test isnt failing on this... may imply we need more owls?

Reply:
tested with another owl. still works

Reply:
Yeah, that was an unused result that happened to be Ok in the test cases, fixed.